File: src/com/train/algorithm/twoPointers/MinimumSizeSubarraySum.cpp
//
// Created by 罗斯 on 23/8/2022.
//
#include <vector>

using namespace std;
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int l = 0,t= 0,r = 0;
        int res = INT_MAX;
        while(l < nums.size()){
            while(t < target && r < nums.size()) t += nums[r++];
            if(t < target) break;
            res = min(res,r - l);
            t -= nums[l++];
        }
        return res == INT_MAX ? 0 : res;
    }
};

File: src/com/train/algorithm/twoPointers/implementInPython/MinimumSizeSubarraySum.py
class Solution(object):
    def minSubArrayLen(self, target, nums):
        i, j, sum, res = 0, 0, 0, 1000000
        while j < len(nums):
            sum += nums[j]

            while sum >= target:
                res = min(res, j - i + 1)
                sum -= nums[i]
                i += 1

            j += 1

        return 0 if res == 1000000 else res